mangarack-component-core/src/providers/dynastyreader/chapter.ts

```diff
@@ -0,0 +1,57 @@
+import * as mio from '../../default';
+import {createPage} from './page';
+const httpService = mio.dependency.get<mio.IHttpService>('IHttpService');
+const htmlService = mio.dependency.get<mio.IHtmlService>('IHtmlService');
+
+/**
+ * Creates the chapter.
+ * @internal
+ * @param address The address.
+ * @param metadata The metadata.
+ * @return The chapter.
+ */
+export function createChapter(address: string, metadata: mio.IChapterMetadata): mio.IChapter {
+  return {
+    number: metadata.number,
+    pagesAsync: () => downloadPagesAsync(address),
+    title: metadata.title,
+    version: metadata.version,
+    volume: metadata.volume
+  };
+}
+
+/**
+ * Promises the document.
+ * @param address The address.
+ * @return The promise for the document.
+ */
+async function downloadDocumentAsync(address: string): Promise<mio.IHtmlDocument> {
+  const body = await httpService().text(address, {}, mio.RequestType.TimeoutWithRetry).getAsync();
+  return htmlService().load(body);
+}
+
+/**
+ * Promises each page.
+ * @param address The address.
+ * @return The promise for each page.
+ */
+async function downloadPagesAsync(address: string): Promise<mio.IPage[]> {
+  const document = await downloadDocumentAsync(address);
+  return getPages(document);
+}
+
+/**
+ * Retrieves each page.
+ * @param $ The selector.
+ * @return Each page.
+ */
+function getPages($: mio.IHtmlDocument): mio.IPage[] {
+  const scriptTag = $('.container ~ script');
+  const pagesArr: Array<string> = scriptTag.html().match(/\{.*?}/g);
+  const address: string = 'http://dynasty-scans.com/';
+
+  return pagesArr.map((pageString, index) => {
+    const page: any = JSON.parse(pageString);
```

Comment:
Don't use `any`. You're expecting a `{image: string}`, so mention this explicitly. This will also ensure a future reader will understand what it is you're expecting to parse there, and if this changes in the future, the new type cast will safe guard additional code (even if it is pretty obvious in this code, it's a good habit to get into).